libs/api/domains/license-service/src/lib/licenseService.service.ts

```diff
@@ -474,4 +476,53 @@ export class LicenseServiceService {
     }
     return verification.data
   }
+
+  async createBarcode(
+    { nationalId }: User,
+    genericUserLicense: GenericUserLicense,
+  ) {
+    const code = randomUUID()
+    const licenseType = genericUserLicense.license.type
+    const client = await this.getClient<typeof licenseType>(licenseType)
+    let extraData: LicenseVerifyExtraDataResult<LicenseType> | undefined
+
+    if (client?.verifyExtraData) {
+      extraData = await client.verifyExtraData({
+        licenseId: genericUserLicense.payload?.metadata?.licenseId,
+        nationalId,
+      })
+
+      if (!extraData) {
+        const msg = `Unable to verify extra data for license: ${licenseType}`
+        this.logger.warn(msg, { category: LOG_CATEGORY })
+
+        throw new InternalServerErrorException(msg)
+      }
+    }
+
+    // Create a token with the version, license type and a server reference (Redis key) code
+    const createTokenPromise = this.tokenService.createToken(
+      {
+        v: '2',
+        t: genericUserLicense.license.type,
```

Comment:
~We might even be able to get rid of the license type from the token and just store it in redis.

Reply:
Agreed removed from token and stored in redis